File: app/src/main/kotlin/com/example/unittesting/login/presenter/LoginView.java
package com.example.unittesting.login.presenter;

public interface LoginView {

    void showProgress();

    void hideProgress();

    void onLoginSuccessful();

    void showLoginError(String errorMessage);

    void showPasswordError(String errorMessage);

    void requestLoginFocus();

    void requestPasswordFocus();
}

File: app/src/main/kotlin/com/example/unittesting/login/presenter/LoginView.kt
package com.example.unittesting.login.presenter

interface LoginView {

    fun showProgress()

    fun hideProgress()

    fun onLoginSuccessful()

    fun showLoginError(errorMessage: String?)

    fun showPasswordError(errorMessage: String?)

    fun requestLoginFocus()

    fun requestPasswordFocus()
}
